modules/lib/shell.nix

```
@@ -1,11 +1,23 @@
 { lib }:
 
-rec {
+let
+  mkShellIntegrationOption = name: default:
+    lib.mkEnableOption "${name} integration" // {
+      inherit default;
+      example = !default;
```

Unfortunately cannot allow `default` or `example` to be affected by the configuration since it would in turn affect the documentation build. Can resolve it by adding a `defaultText` and make the example hard-coded. Something like this (untested) perhaps

```suggestion
      defaultText = literalExpression "config.shell.enableShellIntegration";
      example = false;
```